File: js/examples/exitPosition.js
import 'dotenv/config';
import { pathToFileURL } from 'node:url';
import { EZManagerSDK } from '../sdk.js';

const POSITION_KEY = '';
const SLIPPAGE_PCT = 0.005; // 0.5%

export async function runExitPosition({ sdk, key = POSITION_KEY, slippage = SLIPPAGE_PCT } = {}) {
  if (!key) throw new Error('POSITION_KEY is required');
  const localSdk = sdk ?? (await EZManagerSDK.fromEnv());

  console.log('Exiting position...');
  const result = await localSdk.exitPosition({ keys: [key], slippage });
  console.log(`Position exited! Tx: ${result.txHash}`);
  return { result };
}

const isDirectRun = process.argv[1] && import.meta.url === pathToFileURL(process.argv[1]).href;

if (isDirectRun) {
  await runExitPosition();
}

File: python/examples/exitPosition.py
from pathlib import Path
import sys

from dotenv import load_dotenv

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))
from sdk import EZManagerSDK

load_dotenv()

POSITION_KEY = ''
SLIPPAGE_PCT = 0.005  # 0.5%


def run_exit_position(
    sdk: EZManagerSDK | None = None,
    key: str = POSITION_KEY,
    slippage: float = SLIPPAGE_PCT,
):
    if not key:
        raise RuntimeError('POSITION_KEY is required')

    local_sdk = sdk or EZManagerSDK.from_env()

    print('Exiting position...')
    result = local_sdk.exit_position([key], slippage=slippage)
    print(f"Position exited! Tx: {result['tx_hash']}")
    return {'result': result}


if __name__ == '__main__':
    run_exit_position()